src/main/java/com/google/starfish/servlets/UserRegistrationServlet.java

```diff
@@ -0,0 +1,128 @@
+package com.google.starfish.servlets;  
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import java.io.IOException;  
+import java.io.PrintWriter;  
+import java.sql.Connection;  
+import java.sql.SQLException;  
+import java.sql.PreparedStatement;
+import java.sql.Types;  
+import java.sql.ResultSet;
+import java.sql.Date;
+import javax.servlet.ServletException;  
+import java.security.GeneralSecurityException;
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpSession;  
+import javax.servlet.http.Cookie;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/** Servlet that handles user registration and authentication. */  
+@WebServlet("/user-signin")  
+public class UserRegistrationServlet extends HttpServlet {
+
+  private static final Logger LOGGER = Logger.getLogger(UserRegistrationServlet.class.getName());
+  private static final String CLIENT_ID = System.getenv("CLIENT_ID");
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance())
+      .setAudience(Collections.singletonList(CLIENT_ID))
+      .build();
+
+    try {
+      GoogleIdToken idToken = verifier.verify(req.getParameter("idToken"));
+      if (idToken == null) {
+        res.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        res.getWriter().println("INVALID TOKEN ID");
+        return;
+      }
+
+      Payload payload = idToken.getPayload();
+
+      String userId = payload.getSubject();
+
+      // Get profile information from payload
+      String email = payload.getEmail();
+      String name = (String) payload.get("name");
+      String pictureUrl = (String) payload.get("picture");
+      String locale = (String) payload.get("locale");
+      String familyName = (String) payload.get("family_name");
```

Comment:
What is the purpose of `locale` and `familyName`? It doesn't seem like they're being used anywhere else

Reply:
Good catch, I just extracted everything I could but if they're not being used anywhere then there's no point of assigning a variable